pkg/apis/agent/v1/klusterletaddonconfig_types.go

```diff
@@ -16,84 +16,93 @@ import (
+
+	// ImageRegistry defined the custom registry address of the images.
+	// DEPRECATED in release 2.4 and will be removed in release 2.5 since not used anymore.
+	// +optional
+	ImageRegistry string `json:"imageRegistry,omitempty"`
 
-	ImageRegistry    string `json:"imageRegistry,omitempty"`
-	ImageNamePostfix string `json:"imageNamePostfix,omitempty"`
+	// DEPRECATED in release 2.4 and will be removed in release 2.5 since not used anymore.
 	// +kubebuilder:validation:MinLength=1
 	ImagePullSecret string `json:"imagePullSecret,omitempty"`
 
+	// DEPRECATED in release 2.4 and will be removed in release 2.5 since not used anymore.
 	// +kubebuilder:validation:Enum=Always;Never;IfNotPresent
 	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
-
-	// // ComponentTagMap contains the tag of each component
-	// ComponentTagMap map[string]string `json:"componentTagMap"`
-	// // ComponentImageMap contains the image name of each component
-	// ComponentImageMap map[string]string `json:"componentImageMap"`
-
-	// used for dev work only
-	ComponentOperatorImage string `json:"componentOperatorImage,omitempty"`
 }
 
-// KlusterletAddonConfigApplicationManagerSpec defines configuration for the ApplicationManager component
-type KlusterletAddonConfigApplicationManagerSpec struct {
-	Enabled bool `json:"enabled"`
-
+// ProxyConfig defines the global proxy env for OCP cluster
+type ProxyConfig struct {
+	// HTTPProxy is the URL of the proxy for HTTP requests.  Empty means unset and will not result in an env var.
 	// +optional
-	ArgoCDCluster bool `json:"argocdCluster"`
-}
+	HTTPProxy string `json:"httpProxy,omitempty"`
 
-// KlusterletAddonConfigSearchCollectorSpec defines configuration for the SearchCollector component
-type KlusterletAddonConfigSearchCollectorSpec struct {
-	Enabled bool `json:"enabled"`
-}
+	// HTTPSProxy is the URL of the proxy for HTTPS requests.  Empty means unset and will not result in an env var.
+	// +optional
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
 
-// KlusterletAddonConfigCertPolicyControllerSpec defines configuration for the CertPolicyController component
-type KlusterletAddonConfigCertPolicyControllerSpec struct {
-	Enabled bool `json:"enabled"`
+	// NoProxy is a comma-separated list of hostnames and/or CIDRs for which the proxy should not be used.
+	// Empty means unset and will not result in an env var.
+	// +optional
+	NoProxy string `json:"noProxy,omitempty"`
 }
 
-// KlusterletAddonConfigIAMPolicyControllerSpec defines configuration for the IAMPolicyController component
-type KlusterletAddonConfigIAMPolicyControllerSpec struct {
-	Enabled bool `json:"enabled"`
-}
+type GlobalProxyStatus string
 
-// KlusterletAddonConfigWorkManagerSpec defines configuration for the WorkManager component
-type KlusterletAddonConfigWorkManagerSpec struct {
-	ClusterLabels map[string]string `json:"clusterLabels"`
-}
+const (
+	GlobalProxyStatusTrue  GlobalProxyStatus = "true"
```

Comment:
I think you should have values of Disabled/OCPGlobalProxy/CustomProxy and default is Disabled

Reply:
done